src/bootstrap/test.rs

```diff
@@ -1473,7 +1473,13 @@ note: if you're sure you want to do this, please open an issue as to why. In the
 
         cmd.arg("--src-base").arg(builder.src.join("tests").join(suite));
         cmd.arg("--build-base").arg(testdir(builder, compiler.host).join(suite));
-        cmd.arg("--sysroot-base").arg(builder.sysroot(compiler));
+
+        use std::ops::Deref;
+        let sysroot = match env::var("COMPILETEST_OVERRIDE_SYSROOT") {
+            Ok(sysroot) => PathBuf::from(sysroot),
+            Err(_) => builder.sysroot(compiler).deref().to_owned(),
+        };
+        cmd.arg("--sysroot-base").arg(sysroot);
```

Comment:
So, how should I solve this? Without this, the tests are executed with the stage0 sysroot (logically, because bootstrap/compiletest thinks that I use stage0). But I actually use an external sysroot that I need to pass to tests, otherwise they fail.

Reply:
"stage 0 sysroot" is an overloaded term. There are two relevant sysroots: the one from the bootstrap compiler (`stage0/bin/rustc --print sysroot`) and the one with the newly built std (`stage0-sysroot`). I'm suggesting you change compiletest from using the latter to using the former.

BTW, what are the test failures if you use `stage0-sysroot`?

Reply:
```
--- stderr -------------------------------
error: failed to get bitcode from object file for LTO (Bitcode section not found in object file)

error: aborting due to previous error
------------------------------------------
```
I assume that it might be caused by generating some IR using the stage0 LLVM, as opposed to the current optimized LLVM.

Reply:
ah sorry, i assumed you were putting the PGO compiler in stage 0 - the right sysroot would be build/dist/build-sysroot or wherever the PGO compiler lives.

Reply:
Yes, so that's what I was trying to do with this change - allow overriding the sysroot (which would normally be set to `stage0-sysroot` by bootstrap) with an environment variable, which I set to `build/dist/build-sysroot`.